### ISSUE TITLE
Lazy iteration over points

### ISSUE THREAD

**Issue by ParadaCarleton** [SciML/QuasiMonteCarlo.jl#68]
It's generally a lot better for performance to iterate lazily over QMC points, especially when the number of points is quite large. When users need the whole set, there's nothing stopping them from calling `collect`.

**Comment by ChrisRackauckas** [comment]
Agreed. It would be nice to have an alternative to `sample` which returns an iterator for this.

**Comment by ChrisRackauckas** [comment]
Is https://github.com/SciML/QuasiMonteCarlo.jl/pull/85 sufficient @ParadaCarleton ?

**Comment by dmetivie** [comment]
I think what @ParadaCarleton had in mind is not for randomizing (which #85 do) but rather when sampling. 
Like this is actually done for Sobol points in Sobol.jl with the `next!` function.
